### PR TITLE
Add example of compiling wat and running with wasmtime

### PR DESCRIPTION
**Pull request by AlanFoster**
Adding a simple hello world example of WASI to allow developers without any context to understand WASI at its simplest level.

Let me know your thoughts :+1:

### CODE REVIEW

File: docs/WASI-tutorial.md [hunk 1]
```diff
@@ -244,3 +249,59 @@ The capability model is very powerful, and what's shown here is just the beginni
...
+    ;; Import the required fd_write WASI function which will write the given io vectors to stdout
+    ;; (File Descriptor, *iovs, iovs_len, nwritten) -> Returns number of bytes written
+    (import "wasi_unstable" "fd_write" (func $fd_write (param i32 i32 i32 i32) (result i32)))
+
+    (memory 1)
+    (export "memory" (memory 0))
+
+    ;; Write 'hello world\n' to memory at an offset of 8 bytes
+    ;; Note the trailing newline which is required for the text to appear
+    (data (i32.const 8) "hello word\n")
```
Comment (kubkon) [inline docs/WASI-tutorial.md]:
```suggestion
    (data (i32.const 8) "hello world\n")
```

Comment (kubkon) [reply docs/WASI-tutorial.md]: Looks great to me! Thanks! One thing though, did you know that you don't need to compile WAT to WASM in order to run in wasmtime? You could actually simply run:

```
$ wasmtime demo.wat
```

Perhaps it'd be good to add in the tutorial as well, next to

```
$ wat2wasm demo.wat
```

What do you think?

**Comment by kubkon** [comment]
Looks great to me! Thanks! One thing though, did you know that you don't need to compile WAT to WASM in order to run in wasmtime? You could actually simply run:

```
$ wasmtime demo.wat
```

Perhaps it'd be good to add in the tutorial as well, next to

```
$ wat2wasm demo.wat
```

What do you think?

**Comment by AlanFoster** [comment]
@kubkon Awesome; I've jigged things around to show running `wasmtime` with both `.wat` and `.wasm` files :+1:

**Comment by kubkon** [comment]
LGTM! Thanks again!

**Pull request merged**